src/actions/attack.cpp

```diff
@@ -143,9 +143,9 @@ battle_context_unit_stats::battle_context_unit_stats(const unit& u,
 	petrifies = weapon->bool_ability("petrifies");
 	poisons = !opp.get_state("unpoisonable") && weapon->bool_ability("poison") && !opp.get_state(unit::STATE_POISONED);
 	backstab_pos = is_attacker && backstab_check(u_loc, opp_loc, units, resources::gameboard->teams());
-	rounds = weapon->get_specials("berserk").highest("value", 1).first;
+	rounds = std::max(0, weapon->get_specials("berserk").highest("value", 1).first);
```

Comment:
Do you need to update the other battle_context_unit_stats constructor?

Reply:
@newfrenchy83 Just reminding you that this still needs to happen, both here and in #4122.

Reply:
@newfrenchy83 Please answer my question.

Reply:
i already changed attack.cpp for replace sts::max by clamp in both hre and in https://github.com/wesnoth/wesnoth/pull/4122

Reply:
Yes, but the constructor has two overloads and you only updated one of them.